src/proto/cohort.proto

```diff
@@ -0,0 +1,56 @@
+// API for the cohorts to process transactions from the coordinator.
+
+syntax = "proto3";
+
+package cohort;
+
+import "src/proto/common.proto";
+
+message PrepareTransactionRequest {
+  // Identifier for transaction within cohorts/coordinators. Globally unique
+  // across all requests for all clients. Client can use this to request status
+  // updates and to check the blockchain themselves.
+  string transaction_id = 1;
+  // All operations should have the namespace of the current cohort. Any
+  // operations for other namespaces can be ignored.
+  common.Transaction transaction = 2;
+  // Should be the final config updated by the coordinator.
+  common.TransactionConfig config = 3;
+}
+
+message PrepareTransactionResponse {
+  oneof status {
+    common.PendingResponse pending_response = 1;
+    common.AbortReason aborted_response = 2;
+  }
+}
+
+message GetTransactionResultRequest {
+  // Identifier for transaction within cohorts/coordinators. Globally unique
+  // across all requests for all clients. Client can use this to request status
+  // updates and to check the blockchain themselves.
+  string transaction_id = 1;
+}
+
+message GetTransactionResultResponse {
+  oneof status {
```

Comment:
Does proto3 offer a way to wrap these (oneof ...) into a single field which we can reuse in many places (e.g. as part of the blockchain's response)?

Reply:
I could move it to a common message (e.g. TransactionResult), but I'm not sure we want to do that here since the response here (and in the coordinator) will include the get responses, while the blockchain's response will not. I think it might be best to just include the common fields that are needed.